go/externals/proof_service_generic_social.go

```diff
@@ -0,0 +1,159 @@
+	url = strings.Replace(url, sigHashKey, sigID.String(), 1)
+	if !strings.Contains(url, kbUsername) {
+		return "", fmt.Errorf("Invalid PrefillUrl, missing kbUsername")
+	}
+	if !strings.Contains(url, sigID.String()) {
+		return "", fmt.Errorf("Invalid PrefillUrl, missing sigHash")
+	}
+	return url, nil
+}
+
+//=============================================================================
+
+type GenericSocialProofServiceType struct {
+	libkb.BaseServiceType
+	config *GenericSocialProofConfig
+}
+
+func NewGenericSocialProofServiceType(config *GenericSocialProofConfig) GenericSocialProofServiceType {
+	return GenericSocialProofServiceType{
+		config: config,
+	}
+}
+
+func (t GenericSocialProofServiceType) AllStringKeys() []string { return t.BaseAllStringKeys(t) }
+
+func (t GenericSocialProofServiceType) NormalizeUsername(s string) (string, error) {
+	if !t.config.usernameRe.MatchString(s) {
+		return "", libkb.NewBadUsernameError(s)
+	}
+	// TODO always normalize ToLower?
+	return strings.ToLower(s), nil
+}
+
+func (t GenericSocialProofServiceType) NormalizeRemoteName(m libkb.MetaContext, s string) (ret string, err error) {
+	return t.NormalizeUsername(s)
+}
+
+func (t GenericSocialProofServiceType) GetPrompt() string {
+	return fmt.Sprintf("Your username on %s", t.config.DisplayName)
+}
+
+func (t GenericSocialProofServiceType) ToServiceJSON(username string) *jsonw.Wrapper {
+	return t.BaseToServiceJSON(t, username)
+}
+
+func (t GenericSocialProofServiceType) PostInstructions(username string) *libkb.Markup {
+	return libkb.FmtMarkup(`Please click on the following link to post to %v:`, t.config.DisplayName)
+}
+
+func (t GenericSocialProofServiceType) DisplayName(username string) string {
+	return t.config.DisplayName
+}
+func (t GenericSocialProofServiceType) GetTypeName() string { return t.config.Domain }
+
+func (t GenericSocialProofServiceType) RecheckProofPosting(tryNumber int, status keybase1.ProofStatus, _ string) (warning *libkb.Markup, err error) {
+	return t.BaseRecheckProofPosting(tryNumber, status)
+}
+
+func (t GenericSocialProofServiceType) GetProofType() string {
+	return libkb.GenericSocialWebServiceBinding
+}
+
+func (t GenericSocialProofServiceType) CheckProofText(text string, id keybase1.SigID, sig string) (err error) {
+	// We don't rely only any server trust in FormatProofText so there is nothing to verify here.
+	return nil
+}
+
+func (t GenericSocialProofServiceType) FormatProofText(m libkb.MetaContext, ppr *libkb.PostProofRes,
+	kbUsername string, sigID keybase1.SigID) (string, error) {
+	return t.config.prefillURLWithValues(kbUsername, sigID)
+}
+
+func (t GenericSocialProofServiceType) MakeProofChecker(l libkb.RemoteProofChainLink) libkb.ProofChecker {
+	return &GenericSocialProofChecker{l}
+}
+
+func (t GenericSocialProofServiceType) IsDevelOnly() bool { return false }
+
+// We can post multiple generic social proofs to a single service.
+func (t GenericSocialProofServiceType) LastWriterWins() bool { return false }
```

Comment:
You can only have only live proof for each unique value of `GetTypeName`. Isn't LastWriterWins exactly what you want? Same as github?